src/pip/_internal/cli/cmdoptions.py

```diff
@@ -403,13 +403,17 @@ def _get_format_control(values, option):
 
 
 def _handle_no_binary(option, opt_str, value, parser):
+    if value[0] == '-':
+        raise CommandError("--no-binary requires an argument.")
     existing = _get_format_control(parser.values, option)
     FormatControl.handle_mutual_excludes(
         value, existing.no_binary, existing.only_binary,
     )
 
 
 def _handle_only_binary(option, opt_str, value, parser):
+    if value[0] == '-':
```

Comment:
Add the conditional to the `FormatControl.handle_mutual_excludes` instead.

Reply:
Ah, I thought there might be a problem there! I tried testing with no arg at all following it and it still worked, but I didn't think to try an empty string as an arg.

Reply:
Actually, optparse will catch the mistake before it gets to my check if the user passes in an empty string. However, I will still change my check to use `startswith`.